redux_userlist_06/src/reducers/userReducer.ts

```diff
@@ -0,0 +1,32 @@
+import { UserDetails } from "../interfaces";
+import * as constants from "../constants/constants";
+const initialState: Array<UserDetails> = [];
+
+const UserReducer = (state = initialState, action:{type:string,payload:any}) => {
+  switch (action.type) {
+    case constants.CREATE:
+      state.push(action.payload);
+      return state
```

Comment:
you haven't updated state in immutable way. Make initial state as an object and inside that object store your states as you like.